sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationEventProcessor/CallAutomationEventProcessor.cs

```diff
@@ -62,8 +62,10 @@ public void ProcessEvents(IEnumerable<CallAutomationEventBase> events)
                     handlers(this, args);
                 }
 
-                // If this call is disconnect, remove all related items in memory
-                if (receivedEvent is CallDisconnected)
+                // If call is disconnected, CreateCall failed or Answer failed remove all related items in memory
+                if (receivedEvent is CallDisconnected
+                    || receivedEvent is CreateCallFailed
```

Comment:
we should also add this in CreateCall / AnswerCall event processor result:
See here how it was implemented: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.CallAutomation/src/CallAutomationEventProcessor/EventResult/CreateCallEventResult.cs